pkg/openwsn/doc.txt

```diff
@@ -0,0 +1,413 @@
+/**
+ @defgroup pkg_openwsn   OpenWSN network stack
+ @ingroup  pkg
+ @ingroup  net
+ @brief    Provides a RIOT adaption of the OpenWSN network stack
+ @see      https://github.com/openwsn-berkeley/openwsn-fw
+ @experimental
+
+ # OpenWSN RIOT Port
+
+ This implementation integrates the [OpenWSN](https://github.com/openwsn-berkeley/openwsn-fw)
+ full network stack (UDP, IPv6 (6LoWPAN), RPL, 6TiSCH) into RIOT.
+
+ It can be used instead of GNRC on supported 802.15.4 radios, and compared to
+ GNRC, provides a full 6TiSCH implementation. It does not yet support RIOT's
+ sock API, so applications will have to be written against OpenWSN's API.
+
+ This port provides a new RIOT "board" to the OpenWSN software. In this way
+ RIOT's hardware abstraction connects to OpenWSN's interfaces.
+
+ The simple scheduling mechanism in OpenWSN is run in a RIOT thread with
+ second highest priority after the radio thread (THREAD_PRIORITY_MAIN - 4).
+
+ The current port of OpenWSN currently needs a root node that works along an
+ external tool that performs routing and handles join procedure:
+ [Openvisualizer](https://github.com/openwsn-berkeley/openvisualizer)
+
+
+ ## Joining a network
+
+ The first thing a new mote will need to do is to find a network. On boot it
+ will actively be listening for enhanced beacons. Once a beacon is received it
+ will adjust its timers drift and synchronize with the network. Re-synchronization
+ will be happening constantly to compensate for oscillator and timer drifts.
+
+ Once synchronized the node will need to join the network. OpenWSN uses CoJP
+ [constrained join protocol](https://datatracker.ietf.org/doc/draft-ietf-6tisch-minimal-security/)
+ OpenWSN stack only handles Join Requests. The JRC (join registrar/coordinator,
```

Comment:
The second sentence is weird. Shouldn't it be "OpenWSN uses CoJP
 [constrained join protocol](https://datatracker.ietf.org/doc/draft-ietf-6tisch-minimal-security/)
 and only handles Join Requests" instead ?